packages/streams/src/decorators.ts

```diff
@@ -0,0 +1,18 @@
+import type { Model, Program, Type } from "@typespec/compiler";
+import type { StreamOfDecorator } from "../generated-defs/TypeSpec.Streams.js";
+import { StreamStateKeys } from "./lib.js";
+
+/** @internal */
+export const namespace = "TypeSpec.Streams";
+
+export const $streamOf: StreamOfDecorator = (context, target, type) => {
+  context.program.stateMap(StreamStateKeys.streamOf).set(target, type);
+};
+
+export function getStreamOf(program: Program, target: Model): Type | undefined {
```

Comment:
Not using the useMapState?

Reply:
I hadn't updated to them yet (wrote these before those were checked in) but will try it out!